be/src/exec/pipeline/pipeline_driver.cpp

```diff
@@ -904,4 +905,10 @@ void PipelineDriver::increment_schedule_times() {
     driver_acct().increment_schedule_times();
 }
 
+void PipelineDriver::assign_observer() {
+    for (const auto& op : _operators) {
+        op->set_observer(&_observer);
```

Comment:
All operator must be event observer? it seems that only sink/source operator should be observer?

Reply:
Normally only the assign source and sink operators are needed, but for scenarios where there is a RuntimeFilter, non-source/sink operators need to be able to get this object as well. For example.
HashJoin Build needs to notify the probe side of observe when it finishes, but the probe operator is not a source operator, so we need to set this value to the intermediate operators.